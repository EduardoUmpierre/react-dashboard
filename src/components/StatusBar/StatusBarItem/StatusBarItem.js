import React from 'react'
import './StatusBarItem.scss'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import Loader from '../../Loader/Loader'

const statusBarItem = ({ icon, amount, text, theme }) => {
    const classes = ['status-bar__item', `status-bar__item--${theme}`]

    return (
        <div className='col-12 col-sm-12 col-md-6 col-lg-3'>
            <div className={classes.join(' ')}>
                <div className='status-bar__icon'>
                    <FontAwesomeIcon icon={icon} fixedWidth />
                </div>

                <div className='status-bar__data'>
                    <Loader loaded={amount}>
                        <span className='status-bar__amount'>{amount}</span>
                        <span className='status-bar__text'>{text}</span>
                    </Loader>
                </div>
            </div>
        </div>
    )
}

export default statusBarItem
