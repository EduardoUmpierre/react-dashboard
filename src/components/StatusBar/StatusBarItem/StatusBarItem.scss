.status-bar__item {
    background-color: #fff;
    border-radius: 3px;
    box-shadow: 0 1px 10px 0 rgba(0, 0, 0, 0.05);
    min-height: 5rem;
    overflow: hidden;

    &,
    > .status-bar__icon {
        align-items: center;
        display: flex;
        height: 100%;
    }

    // Icon
    > .status-bar__icon {
        background-color: blue;
        justify-content: center;
        min-width: 6.875rem;

        > svg {
            color: #fff;
            font-size: 2.5rem;
        }
    }

    // Data
    > .status-bar__data {
        display: flex;
        flex-direction: column;
        padding: 0 1.25rem;

        > span {
            line-height: 1;
        }

        > .status-bar__amount {
            color: #64686c;
            font-size: 1.75rem;
            margin-bottom: 0.313rem;
        }

        > .status-bar__text {
            color: #9fadc3;
            font-size: 0.875rem;
        }
    }

    // Modifiers
    &--blue {
        
        > .status-bar__icon {
            background-color: #30a5ff;
        }
    }

    &--yellow {
        
        > .status-bar__icon {
            background-color: #ffb53e;
        }
    }

    &--green {
        
        > .status-bar__icon {
            background-color: #1ebfae;
        }
    }

    &--red {
        
        > .status-bar__icon {
            background-color: #f9243f;
        }
    }
}
