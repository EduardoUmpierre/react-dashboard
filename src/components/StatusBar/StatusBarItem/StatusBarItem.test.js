import React from 'react'
import { shallow } from 'enzyme'
import StatusBarItem from './StatusBarItem'

describe('<StatusBarItem />', () => {
    it('renders correctly', () => {
        const component = shallow(
            <StatusBarItem
                theme='blue'
                icon='shopping-bag'
                text='New Orders'
                amount='1000'
            />
        )

        expect(component).toMatchSnapshot()
    })
})
