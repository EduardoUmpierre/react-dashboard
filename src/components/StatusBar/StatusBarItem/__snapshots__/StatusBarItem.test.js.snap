// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StatusBarItem /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <statusBarItem
    amount="1000"
    icon="shopping-bag"
    text="New Orders"
    theme="blue"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="status-bar__item status-bar__item--blue"
      >
        <div
          className="status-bar__icon"
        >
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={true}
            flip={null}
            icon="shopping-bag"
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            title=""
            transform={null}
          />
        </div>
        <div
          className="status-bar__data"
        >
          <loader
            loaded="1000"
          >
            <span
              className="status-bar__amount"
            >
              <NumericLabel
                params={
                  Object {
                    "locales": "en-US",
                    "precision": 0,
                    "shortFormat": true,
                    "shortFormatMinValue": 1000,
                    "shortFormatPrecision": 1,
                  }
                }
              >
                1000
              </NumericLabel>
            </span>
            <span
              className="status-bar__text"
            >
              New Orders
            </span>
          </loader>
        </div>
      </div>,
      "className": "col-12 col-sm-12 col-md-6 col-lg-3",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="status-bar__icon"
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={true}
              flip={null}
              icon="shopping-bag"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
          </div>,
          <div
            className="status-bar__data"
          >
            <loader
              loaded="1000"
            >
              <span
                className="status-bar__amount"
              >
                <NumericLabel
                  params={
                    Object {
                      "locales": "en-US",
                      "precision": 0,
                      "shortFormat": true,
                      "shortFormatMinValue": 1000,
                      "shortFormatPrecision": 1,
                    }
                  }
                >
                  1000
                </NumericLabel>
              </span>
              <span
                className="status-bar__text"
              >
                New Orders
              </span>
            </loader>
          </div>,
        ],
        "className": "status-bar__item status-bar__item--blue",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={true}
              flip={null}
              icon="shopping-bag"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />,
            "className": "status-bar__icon",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "border": false,
              "className": "",
              "fixedWidth": true,
              "flip": null,
              "icon": "shopping-bag",
              "inverse": false,
              "listItem": false,
              "mask": null,
              "pull": null,
              "pulse": false,
              "rotation": null,
              "size": null,
              "spin": false,
              "symbol": false,
              "title": "",
              "transform": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <loader
              loaded="1000"
            >
              <span
                className="status-bar__amount"
              >
                <NumericLabel
                  params={
                    Object {
                      "locales": "en-US",
                      "precision": 0,
                      "shortFormat": true,
                      "shortFormatMinValue": 1000,
                      "shortFormatPrecision": 1,
                    }
                  }
                >
                  1000
                </NumericLabel>
              </span>
              <span
                className="status-bar__text"
              >
                New Orders
              </span>
            </loader>,
            "className": "status-bar__data",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <span
                  className="status-bar__amount"
                >
                  <NumericLabel
                    params={
                      Object {
                        "locales": "en-US",
                        "precision": 0,
                        "shortFormat": true,
                        "shortFormatMinValue": 1000,
                        "shortFormatPrecision": 1,
                      }
                    }
                  >
                    1000
                  </NumericLabel>
                </span>,
                <span
                  className="status-bar__text"
                >
                  New Orders
                </span>,
              ],
              "loaded": "1000",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <NumericLabel
                    params={
                      Object {
                        "locales": "en-US",
                        "precision": 0,
                        "shortFormat": true,
                        "shortFormatMinValue": 1000,
                        "shortFormatPrecision": 1,
                      }
                    }
                  >
                    1000
                  </NumericLabel>,
                  "className": "status-bar__amount",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": "1000",
                    "params": Object {
                      "locales": "en-US",
                      "precision": 0,
                      "shortFormat": true,
                      "shortFormatMinValue": 1000,
                      "shortFormatPrecision": 1,
                    },
                  },
                  "ref": null,
                  "rendered": "1000",
                  "type": [Function],
                },
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "New Orders",
                  "className": "status-bar__text",
                },
                "ref": null,
                "rendered": "New Orders",
                "type": "span",
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="status-bar__item status-bar__item--blue"
        >
          <div
            className="status-bar__icon"
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={true}
              flip={null}
              icon="shopping-bag"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
          </div>
          <div
            className="status-bar__data"
          >
            <loader
              loaded="1000"
            >
              <span
                className="status-bar__amount"
              >
                <NumericLabel
                  params={
                    Object {
                      "locales": "en-US",
                      "precision": 0,
                      "shortFormat": true,
                      "shortFormatMinValue": 1000,
                      "shortFormatPrecision": 1,
                    }
                  }
                >
                  1000
                </NumericLabel>
              </span>
              <span
                className="status-bar__text"
              >
                New Orders
              </span>
            </loader>
          </div>
        </div>,
        "className": "col-12 col-sm-12 col-md-6 col-lg-3",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="status-bar__icon"
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={true}
                flip={null}
                icon="shopping-bag"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
            </div>,
            <div
              className="status-bar__data"
            >
              <loader
                loaded="1000"
              >
                <span
                  className="status-bar__amount"
                >
                  <NumericLabel
                    params={
                      Object {
                        "locales": "en-US",
                        "precision": 0,
                        "shortFormat": true,
                        "shortFormatMinValue": 1000,
                        "shortFormatPrecision": 1,
                      }
                    }
                  >
                    1000
                  </NumericLabel>
                </span>
                <span
                  className="status-bar__text"
                >
                  New Orders
                </span>
              </loader>
            </div>,
          ],
          "className": "status-bar__item status-bar__item--blue",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={true}
                flip={null}
                icon="shopping-bag"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />,
              "className": "status-bar__icon",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "border": false,
                "className": "",
                "fixedWidth": true,
                "flip": null,
                "icon": "shopping-bag",
                "inverse": false,
                "listItem": false,
                "mask": null,
                "pull": null,
                "pulse": false,
                "rotation": null,
                "size": null,
                "spin": false,
                "symbol": false,
                "title": "",
                "transform": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <loader
                loaded="1000"
              >
                <span
                  className="status-bar__amount"
                >
                  <NumericLabel
                    params={
                      Object {
                        "locales": "en-US",
                        "precision": 0,
                        "shortFormat": true,
                        "shortFormatMinValue": 1000,
                        "shortFormatPrecision": 1,
                      }
                    }
                  >
                    1000
                  </NumericLabel>
                </span>
                <span
                  className="status-bar__text"
                >
                  New Orders
                </span>
              </loader>,
              "className": "status-bar__data",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <span
                    className="status-bar__amount"
                  >
                    <NumericLabel
                      params={
                        Object {
                          "locales": "en-US",
                          "precision": 0,
                          "shortFormat": true,
                          "shortFormatMinValue": 1000,
                          "shortFormatPrecision": 1,
                        }
                      }
                    >
                      1000
                    </NumericLabel>
                  </span>,
                  <span
                    className="status-bar__text"
                  >
                    New Orders
                  </span>,
                ],
                "loaded": "1000",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <NumericLabel
                      params={
                        Object {
                          "locales": "en-US",
                          "precision": 0,
                          "shortFormat": true,
                          "shortFormatMinValue": 1000,
                          "shortFormatPrecision": 1,
                        }
                      }
                    >
                      1000
                    </NumericLabel>,
                    "className": "status-bar__amount",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": "1000",
                      "params": Object {
                        "locales": "en-US",
                        "precision": 0,
                        "shortFormat": true,
                        "shortFormatMinValue": 1000,
                        "shortFormatPrecision": 1,
                      },
                    },
                    "ref": null,
                    "rendered": "1000",
                    "type": [Function],
                  },
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "New Orders",
                    "className": "status-bar__text",
                  },
                  "ref": null,
                  "rendered": "New Orders",
                  "type": "span",
                },
              ],
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
